### ISSUE TITLE
IntelliJ 2024.2: ProcessCanceledException class hierarchy changed. Resolve issues that arise from this API change.

### ISSUE THREAD

**Issue by mrglavas** [OpenLiberty/liberty-tools-intellij#767]
Change in IntelliJ 2024.2 https://github.com/JetBrains/intellij-community/commit/12e78f18d6352502c7cc942dc677f02bb9b8b257#diff-53953a080bfbb7f3d529336a7318cfde6b06098f5fbb136f7c58c3fad831f500
See resolution in LSP4iJ: https://github.com/redhat-developer/lsp4ij/pull/293.

LTI has multiple places with multi catch of `CancellationException` and `ProcessCanceledException` and will face the same problem.

LSP4iJ compilation errors that occurred with IntelliJ 2024.2:
![image](https://github.com/OpenLiberty/liberty-tools-intellij/assets/890521/409ace29-bf66-43a1-b252-c113cd2d43e1)

**Comment by vaisakhkannan** [comment]
Usages of the following
catch (IndexNotReadyException | ProcessCanceledException | CancellationException e)

- [ ] src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/AnnotationUtil.getScopeAnnotations()
- [ ] src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/annotations/PostConstructReturnTypeQuickFix.resolveCodeAction()
- [ ] src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/beanvalidation/BeanValidationQuickFix.resolveRemoveConstraintAnnotationsCodeAction()
- [ ] src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/beanvalidation/BeanValidationQuickFix.resolveStaticModifierCodeAction()
- [ ] src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/cdi/ManagedBeanNoArgConstructorQuickFix.resolveCodeAction()
- [ ] src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/cdi/ManagedBeanQuickFix.resolveCodeAction()
- [ ] src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/codeAction/proposal/quickfix/InsertAnnotationMissingQuickFix.resolveCodeAction()
- [ ] src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/codeAction/proposal/quickfix/InsertAnnotationQuickFix.resolveCodeAction()
- [ ] src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/codeAction/proposal/quickfix/RemoveAnnotationConflictQuickFix.resolveCodeAction()
- [ ] src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/codeAction/proposal/quickfix/RemoveMethodParametersQuickFix.resolveCodeAction()
- [ ] src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/codeAction/proposal/quickfix/RemoveModifierConflictQuickFix.resolveCodeAction()
- [ ] src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/codeAction/proposal/quickfix/RemoveParamAnnotationQuickFix.resolveCodeAction()
- [ ] src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/jax_rs/NoResourcePublicConstructorQuickFix.convertWorkspaceEdit()
- [ ] src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/jax_rs/NonPublicResourceMethodQuickFix.resolveCodeAction()

**Comment by mrglavas** [comment]
@vaisakhkannan I've moved this back to parked. We're not targeting IntelliJ 2024.2 for 24.0.8. Will revisit when we do move up to 2024.2.

**Comment by vaisakhkannan** [comment]
The changes in the above PR have been merged into the feature branch Support_intellij_2024.2.

**Comment by mrglavas** [comment]
@vaisakhkannan I noticed the PR was merged on to the feature branch without a code review. Adding review comments here about the code that was changed.

It would be desirable to refactor out common code so that the next time the exception handling requires an update, it won't require changes in so many places. For instance, this could be accomplished by writing a method which accepts a function (i.e. a lambda expression) as a parameter. The method would call the function which is wrapped in a try/catch. If the exception handling needs to be updated again then you only need to update the code in one place. In general if you find that you're making the same change in multiple places it could be a sign of duplicated code which could be improved.
